src/naming.jl

```diff
@@ -250,3 +250,89 @@ end
 function dict2ntuple(dict::Dict{Symbol, T}) where T
     NamedTuple{Tuple(keys(dict))}(values(dict))
 end
+
+
+"""
+    parse_savename(filename::AbstractString; kwargs...)
+                        parsetypes = (Int, Float64),
+                        connector = "_")
+Try to convert a shorthand name produced with [`savename`](@ref) into a dictionary 
+containing the parameters, a prefix and suffix string.
+
+## Keywords
+* `connector = "_"` : string used to connect the various entries.
+* `parsetypes = (Int, Float64)` : tuple used to define the types which should
+  be tried when parsing the values given in `filename`. Fallback is `String`.
+"""
+function parse_savename(filename::AbstractString;
+                        parsetypes = (Int, Float64),
+                        connector = "_")
+    # Prefix can be anything, so it might also contain a folder which's
+    # name was generated using savename. Therefore first the path is split
+    # into folders and filename.
+    prefix_part, savename_part = dirname(filename),basename(filename)
+    # Extract the suffix. A suffix is identified by searching for the last "."
+    # after the last "=".
+    last_eq = findlast("=",savename_part)
+    last_dot = findlast(".",savename_part)
+    if last_dot == nothing || last_eq > last_dot
+        # if no dot is after the last "="
+        # there is no suffix
+        name, suffix = savename_part,""
+    else
+        # Check if the last dot is part of a float number by parsing it as Int
+        if tryparse(Int,savename_part[first(last_dot)+1:end]) == nothing
+            # no int, so the part after the last dot is the suffix
+            name, suffix = savename_part[1:first(last_dot)-1], savename_part[first(last_dot)+1:end]
+        else
+            # no suffix, because the dot just denotes the decimal places.
+            name, suffix = savename_part, ""
+        end
+    end
+    # Extract the prefix by searching for the first connector that comes before
+    # an "=".
+    first_eq = findfirst("=",name)
+    first_connector = findfirst(connector,name)
+    if first_connector == nothing || first(first_eq) < first(first_connector)
+        prefix, _parameters = "", name
+    else
+        # There is a connector symbol before, so there might be a connector.
+        # Of course the connector symbol could also be part of the variable name.
+        prefix, _parameters = name[1:first(first_connector)-1], name[first(first_connector)+1:end]
+    end
+    # Add leading directory back to prefix
+    prefix = joinpath(prefix_part,prefix)
+    parameters = Dict{String,Any}()
+    # Regex that matches smalles possible range between $connector and =.
+    # This way it is possible to corretly match something like
+    # string_with_underscore_a=10.1 as the next match for the regex is
+    # string_with[_underscore_a=]10.1
+    name_seperator = Regex("$connector[^$connector]+=")
```

Comment:
Again, this is excellent; I would never have been able to do this myself, since I don't really know much about Regex... Jonas knows much more, I assume he would be even happier than me to see this!